maroon/src/app.rs

```diff
@@ -1,22 +1,55 @@
 use crate::p2p_interface::{Inbox, NodeState, Outbox, P2PChannels};
 use common::{
+    async_interface::{AsyncInterface, ReqResPair},
     gm_request_response::Transaction,
     range_key::{self, KeyOffset, KeyRange, TransactionID},
 };
 use libp2p::PeerId;
-use log::{debug, error, info, warn};
+use log::{error, info};
 use std::{
     collections::{HashMap, HashSet},
     num::NonZeroUsize,
     time::Duration,
 };
 use tokio::sync::oneshot;
 
-pub struct App {
+pub enum Request {
+    GetState,
+}
+#[derive(Debug, PartialEq, Eq)]
+pub enum Response {
+    State(CurrentOffsets),
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct CurrentOffsets {
+    pub self_offsets: HashMap<KeyRange, KeyOffset>,
+    pub consensus_offset: HashMap<KeyRange, KeyOffset>,
```

Comment:
Perhaps keep track of all offsets here in this data structure too? Or do you think they will be too large?

If we can keep them here, the code will (I think) become clearer, since maintaining the consensus offsets map will be a separate module, tiny, self-contained, totally algorithmic, with a fixed set of parameters, tested through independently. I'd vote for that if you considered this option.